minisom.py

```diff
@@ -538,6 +537,28 @@ def topographic_error(self, data):
         dxdy = hstack([diff(b2mu_x), diff(b2mu_y)])
         distance = norm(dxdy, axis=1)
         return (distance > t).mean()
+    
+    def topographic_error_hexagonal(self, data):
+        """Return the topographic error for hexagonal grid"""
+        b2mu_inds = argsort(self._distance_from_weights(data), axis=1)[:, :2]
+        b2mu_coords = [[self.get_euclidean_coordinates_from_index(bmu[0]), 
+                        self.get_euclidean_coordinates_from_index(bmu[1])] 
+                            for bmu in b2mu_inds]
+        b2mu_coords = array(b2mu_coords)
+        b2mu_neighbors = [(bmu1 >= bmu2-1) & ((bmu1 <= bmu2+1)) 
+                            for bmu1, bmu2 in b2mu_coords]
+        b2mu_neighbors = [neighbors.prod() for neighbors in b2mu_neighbors]
+        te = 1 - mean(b2mu_neighbors)
+        return te
+        
+    def get_euclidean_coordinates_from_index(self, index):
```

Comment:
In this repository private methods start with an underscore.

Reply:
Updated